lib/util.js

```diff
@@ -672,222 +644,209 @@ function formatError(value) {
 
 
 function formatObject(ctx, value, recurseTimes, visibleKeys, keys) {
-  return keys.map(function(key) {
-    return formatProperty(ctx, value, recurseTimes, visibleKeys, key, false);
-  });
+  const len = keys.length;
+  const output = new Array(len);
+  for (var i = 0; i < len; i++)
+    output[i] = formatProperty(ctx, value, recurseTimes, visibleKeys, keys[i],
+                               0);
+  return output;
 }
 
 
 function formatArray(ctx, value, recurseTimes, visibleKeys, keys) {
-  var output = [];
-  let visibleLength = 0;
-  let index = 0;
-  for (const elem of keys) {
-    if (visibleLength === ctx.maxArrayLength)
+  const maxLength = ctx.maxArrayLength < 0 ? 0 : ctx.maxArrayLength;
+  const output = [];
+  var visibleLength = 0;
+  var i = 0;
+  for (const key of keys) {
+    if (visibleLength === maxLength)
       break;
-    // Symbols might have been added to the keys
-    if (typeof elem !== 'string')
-      continue;
-    const i = +elem;
-    if (index !== i) {
+    const index = +key;
+    if (i !== index) {
       // Skip zero and negative numbers as well as non numbers
-      if (i > 0 === false)
-        continue;
-      const emptyItems = i - index;
+      if (index > 0 === false)
+        break;
+      const emptyItems = index - i;
       const ending = emptyItems > 1 ? 's' : '';
       const message = `<${emptyItems} empty item${ending}>`;
       output.push(ctx.stylize(message, 'undefined'));
-      index = i;
-      if (++visibleLength === ctx.maxArrayLength)
+      i = index;
+      if (++visibleLength === maxLength)
         break;
     }
-    output.push(formatProperty(ctx, value, recurseTimes, visibleKeys,
-                               elem, true));
+    output.push(formatProperty(ctx, value, recurseTimes, visibleKeys, key, 1));
     visibleLength++;
-    index++;
+    i++;
   }
-  if (index < value.length && visibleLength !== ctx.maxArrayLength) {
-    const len = value.length - index;
+  if (i < value.length && visibleLength !== maxLength) {
+    const len = value.length - i;
     const ending = len > 1 ? 's' : '';
     const message = `<${len} empty item${ending}>`;
     output.push(ctx.stylize(message, 'undefined'));
-    index = value.length;
+    i = value.length;
   }
-  var remaining = value.length - index;
+  const remaining = value.length - i;
   if (remaining > 0) {
     output.push(`... ${remaining} more item${remaining > 1 ? 's' : ''}`);
   }
-  for (var n = 0; n < keys.length; n++) {
-    var key = keys[n];
-    if (typeof key === 'symbol' || !numbersOnlyRE.test(key)) {
-      output.push(formatProperty(ctx, value, recurseTimes, visibleKeys,
-                                 key, true));
+  for (i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    // Skip positive numbers
+    if (+key >= 0 === false) {
```

Comment:
I like the short versions but I try to write more explicit JS and the latter does not work for keys other then numbers. So if you do not feel strongly about it, I would just keep it as is?